README.md

```diff
@@ -0,0 +1,53 @@
+### Example of integrating with inner blocks in the WooCommerce Blocks Checkout
+
+This repository contains some example code showing how an extension can register an inner block for use in the Checkout
+Block offered by WooCommerce Blocks.
+
+To integrate a front-end block with WooCommerce Blocks that extends the API, work is required in two areas:
+Server-side code that extends the `StoreApi` and front-end code that renders and controls your block.
+
+The overview of how this works is as follows:
+
+#### PHP
+
+- The extension sets up its integration with WooCommerce Blocks by extending the `IntegrationInterface` class:
+https://github.com/woocommerce/newsletter-test/blob/main/woocommerce-blocks-integration.php. This class must implement the following methods:
+  - `initialize` - called by WooCommerce Blocks to begin the execution of the integration code.
+  - `get_name` - the name of your extension, this should be a unique identifier for your extension.
+  - `get_script_handles` - Returns an array containing the handles of any scripts registered by our extension.
+  - `get_editor_script_handles` - Returns an array containing the handles of any _editor_ scripts registered by our
+extension.
+  - `get_script_data` - Returns an associative array containing any data we want to be available to the scripts on the
+front-end. You can get to it on the front end by accessing `wcSettings['newsletter-test_data']` (The key on `wcSettings`
+will be the value returned by `get_name` and `_data`).
+- The extension registers this integration when the `woocommerce_blocks_checkout_block_registration` action is fired:
+https://github.com/woocommerce/newsletter-test/blob/main/newsletter-test.php#L26-L33
+- The `initialize` method calls  `register_frontend_scripts()`, `$this->register_editor_scripts()`, and
+`$this->register_editor_blocks()` which registers all the scripts with WordPress. (These will be enqueued later by WooCommerce Blocks, we tell Blocks about the scrips in the `get_script_handles` etc. methods).
+- Next, in the `initialize` method, the extension
+[extends the API schema](https://github.com/woocommerce/newsletter-test/blob/main/woocommerce-blocks-integration.php#L141).
+  - Using the `register_endpoint_data` method on `ExtendSchema` the extension causes the cart endpoint to return
+additional data. More information on this can be found at this URL: https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/extensibility/extend-rest-api-add-data.md
+- Moving
```

Comment:
Oops, un unfinished thought here...

Reply:
Oh 😖 yeah I meant to delete that, thanks for spotting!